### PR TITLE
BENCH-1259: Make App-related support resources optional in schema.

### PR DESCRIPTION
**Pull request by jmczerk**
This makes the application-specific support resources added in v0.6.0 optional in the schema.  Additionally, this has the side effect of making this version backward compatible with v0.5.

While it has not been our intent to permanently maintain backward compatibility, it's useful for our current deployment situation.  As such, this PR also adds a backward compatibility test against data written with the v0.5 schema so that we can detect if/when this backward compatibility is broken, and make an intentional decision regarding breaking backward compatibility at that time.

### CODE REVIEW

**Review by dexamundsen — APPROVED** [review]

File: src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java [hunk 1]
```diff
@@ -268,4 +268,84 @@ public void addFieldBeforeSchemaUpdateTestLogic(EnvironmentDiscovery environment
       throws IOException {
     noLandingZonesTestLogic(environmentDiscovery);
   }
+
+  private void expectMissingLandingZoneOptionals(
+      LandingZone expectedLandingZone, LandingZone landingZone) {
+    Assertions.assertTrue(landingZone.getApplicationVpcId().isEmpty());
+    Assertions.assertTrue(landingZone.getApplicationVpcId().isEmpty());
```
Comment (nfoen) [inline src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java:275]: was this second one meant to be the private subnet id?

File: src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java [hunk 2]
```diff
@@ -268,4 +268,84 @@ public void addFieldBeforeSchemaUpdateTestLogic(EnvironmentDiscovery environment
...
+  public Path getAppsDisabledTestDataPath() {
+    return basePath.resolve(getAppsDisabledTestDataBucketName());
+  }
+
+  /**
+   * Test data contains latest schema with optional Application Support Resources not present.
+   * Verifies that parsing with current schema as "reader" schema succeeds, and that getting these
+   * resource returns empty optionals.
+   */
+  public void appsDisabledTTestLogic(EnvironmentDiscovery environmentDiscovery) throws IOException {
```
Comment (nfoen) [inline src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java:331]: possible misspelling "TTest"

**Review by nfoen — APPROVED** [review]
lgtm

File: src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java (hunk 1 above)
Comment (jmczerk) [reply src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java]: Thank you!

File: src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java (hunk 2 above)
Comment (jmczerk) [reply src/test/java/bio/terra/aws/resource/discovery/EnvironmentDiscoveryTestBase.java]: Thank you!

**Pull request merged**